src/ampState.js

```diff
@@ -0,0 +1,10 @@
+export default function ampState(data) {
+  if (!data) return '';
+  return Object.entries(data).map(([key, value]) => (
+    `<amp-state id="${key}">
```

Comment:
(The linked spec is entirely non-normative, which means browsers aren't required to implement it, which means it's not cross-browser html - what id call "valid")